astropy/table/table.py

```diff
@@ -1342,11 +1342,12 @@ def _convert_data_to_col(
             try:
                 col = data[0].__class__(data)
                 col.info.name = name
```

Comment:
@mhvk, does the `col.info.name` assignment ever trip the `Exception`? If not we can move it to the `else` block

Reply:
Agreed the `col.info.name` should be moved out. If that fails then the `col` object would be malformed. Instead an exception should raised here at the point of the problem.

Reply:
Should still be moved to `else` clause; otherwise this is no improvement, since one is hiding that the `except` clause passes on to the rest of the code.